### PR TITLE
Update Developer Guide for Debugging and Add GDB Pretty Printer

### PR DESCRIPTION
**Pull request by kmurray**
Various updates to documentation on development best with respect to debugging.

Also adds a pretty printer for some VTR data types under `dev/`

### CODE REVIEW

**Comment by mithro** [comment]
We should figure out if we can automatically reference the debugging scripts using https://sourceware.org/gdb/onlinedocs/gdb/dotdebug_005fgdb_005fscripts-section.html#dotdebug_005fgdb_005fscripts-section

File: dev/vtr_pretty_printers.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import re
```
Comment (mithro) [inline dev/vtr_pretty_printers.py:1]: Add a module docstring and link to the using documentation?

File: doc/src/vpr/command_line_usage.rst [hunk 2]
```diff
@@ -1410,3 +1410,15 @@ The following options are used to enable power estimation in VPR.
...
+Command-line Auto Completion
+----------------------------
+
+To simplify using VPR on the command-line you can use the ``dev/vpr_bash_completion.sh`` script, which will enable TAB completion for VPR commandline arguments (based on the output of `vpr -h`).
+
+Simply add:
+
+.. code-block:: bash
+
+    source $VTR_ROOT/dev/vpr_bash_completion.sh
```
Comment (mithro) [inline doc/src/vpr/command_line_usage.rst:1422]: That is cool!

File: README.developers.md [hunk 3]
```diff
@@ -751,6 +751,77 @@ $ make
...
+    from libstdcxx.v6.printers import register_libstdcxx_printers
+    register_libstdcxx_printers(None)
+
+    end
+
+where `$STL_PRINTER_ROOT` should be replaced with the appropriate path to the STL pretty printers.
+For example recent versions of GCC include these under `/usr/share/gcc-*/python` (e.g. `/usr/share/gcc-9/python`)
+
+
+### VTR Pretty Printers
```
Comment (mithro) [inline README.developers.md:791]: This is super cool too!

**Comment by mithro** [comment]
@kmurray This is awesome!

File: dev/vtr_pretty_printers.py (hunk 1 above)
Comment (kmurray) [reply dev/vtr_pretty_printers.py]: Done!

**Comment by kmurray** [comment]
> We should figure out if we can automatically reference the debugging scripts using https://sourceware.org/gdb/onlinedocs/gdb/dotdebug_005fgdb_005fscripts-section.html#dotdebug_005fgdb_005fscripts-section

That sounds like good future work (but beyond the scope of this PR). I'd suggest filing an issue on it.

**Pull request merged**